Library/CorrelationIdManager.ts

```diff
@@ -0,0 +1,93 @@
+import https = require('https');
+import http = require('http');
+import url = require('url');
+
+class CorrelationIdManager {
+    public static correlationIdPrefix: "cid-v1:";
+
+    private static pendingLookups: {[key: string]: Function[]} = {};
+    private static completedLookups: {[key: string]: string} = {};
+
+    public static queryCorrelationId(endpointBase: string, instrumentationKey: string, correlationIdRetryInterval: number, callback: (correlationId: string) => void) {

```

Comment:
`queryCorrelationId` is called when a `Config` is constructed, or when the `profileQueryEndpoint` on a `Config` is changed. A `Config` is constructed when a `Client` is constructed, and you can construct clients explicitly via `appInsights.getClient("key")` if you want.